core/trino-main/src/main/java/io/trino/sql/planner/ExpressionInterpreter.java

```diff
@@ -302,7 +305,9 @@ private Object processWithExceptionHandling(Expression expression, Object contex
             }
 
             try {
-                return process(expression, context);
+                return optimizationCache.computeIfAbsent(
```

Comment:
you must have specific path in `visitInPredicate`, e.g: when only `io.trino.sql.tree.InPredicate#value` changes but not `io.trino.sql.tree.InPredicate#valueList`